core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java

```diff
@@ -155,34 +158,32 @@ public void executeProductTestPlanTest() throws TestGridException, UnsupportedPr
         Mockito.when(productTestPlan.getStatus()).thenReturn(ProductTestPlan.Status.COMPLETED);
 
         InfrastructureProvider provider = Mockito.mock(OpenStackProvider.class);
-        Deployment deployment = Mockito.mock(Deployment.class);
+
         DeployerServiceImpl deployer = mock(DeployerServiceImpl.class);
         TestEngineImpl testEngine = mock(TestEngineImpl.class);
         TestReportEngineImpl testReportEngine = mock(TestReportEngineImpl.class);
 
-
         PowerMockito.mockStatic(InfrastructureProviderFactory.class);
 
         Mockito.when(InfrastructureProviderFactory.getInfrastructureProvider(infrastructure)).thenReturn(provider);
         Mockito.when(provider.createInfrastructure(infrastructure, scenarioLocation)).thenReturn(deployment);
+        Mockito.when(provider.removeInfrastructure(infrastructure, scenarioLocation)).thenReturn(true);
         Mockito.when(provider.canHandle(infrastructure)).thenReturn(true);
         Mockito.when(provider.removeInfrastructure(infrastructure, scenarioLocation)).thenReturn(true);
 
+        Mockito.when(deployer.deploy(testPlan.getDeployment())).thenReturn(deployment);
+        Mockito.when(testEngine.runScenario(testScenario, scenarioLocation, deployment)).thenReturn(true);
 
         try {
-            whenNew(DeployerServiceImpl.class).withNoArguments().thenReturn(deployer);
-            Mockito.when(deployer.deploy(testPlan)).thenReturn(deployment);
-
-            whenNew(TestEngineImpl.class).withNoArguments().thenReturn(testEngine);
-            Mockito.when(testEngine.runScenario(testScenario, scenarioLocation, deployment)).thenReturn(true);
-
-            whenNew(TestReportEngineImpl.class).withNoArguments().thenReturn(testReportEngine);
+            PowerMockito.whenNew(DeployerServiceImpl.class).withNoArguments().thenReturn(deployer);
+            PowerMockito.whenNew(TestEngineImpl.class).withNoArguments().thenReturn(testEngine);
+            PowerMockito.whenNew(TestReportEngineImpl.class).withNoArguments().thenReturn(testReportEngine);
```

Comment:
Why are we catching the exception? If the exception is thrown it means that the test should fail right?

Reply:
Will fix with #54